modules/bday.js

```diff
@@ -38,216 +50,245 @@ const confirmChannel = (response_url, cmd, channel) => (
-            },
-            {
-              'type': 'context',
-              'elements': [
-                {
-                  'type': 'plain_text',
-                  'text': `Card link: ${MIRO_URL}`,
-                  'emoji': true
-                }
-              ]
-            }
-            
-          ]
+          blocks: confirmation
         })
-      })
-      .catch((e) => {
+      }).catch((e) => {
         return axios.post(response_url, {
           response_type: 'ephemeral',
           text: `Something went wrong: ${e}.`,
         })
       })
+    }
   }
 
   // send card to bday user
-  if (command.includes('send') && command.includes('to')) {
-    const R = /<(?<MIRO_URL>.*)> to <@(?<BDAY_USER_ID>.*)\|(?<BDAY_USER_NAME>.*)>/
-    const RwithMessage = /<(?<MIRO_URL>.*)> to <@(?<BDAY_USER_ID>.*)\|(?<BDAY_USER_NAME>.*)> (?<CUSTOM_MESSAGE>.*)/
-    const matches = command.match(R)
-    const matchesWithOptMessage = command.match(RwithMessage)
-
-    // notify user of invalid input
-    if (!matches) {
-      return invalidInputNotice(response_url, command)
+  if (command === 'send') {
+    // first iteration returns modal
+    if (!type) {
+      const state = {
+        response_url,
+        ref: 1,
+        command: 'send',
+      }
+      return web.views.open({
+        trigger_id,
+        view: {
+          'callback_id': 'bday',
+          'private_metadata': JSON.stringify(state),
+          'type': 'modal',
+          'title': {
+            'type': 'plain_text',
+            'text': 'Send bday card'
+          },
+          'blocks': [..._blocks(state.ref), customMessage(state.ref), button],
+          'close': {
+            'type': 'plain_text',
+            'text': 'Nevermind'
+          },
+          'submit': {
+            'type': 'plain_text',
+            'text': 'Send'
+          },
+        }
+      })
     }
-    // send to bday user
-    const { groups: { MIRO_URL, BDAY_USER_ID } } = matches
-    const sender = await web.users.info({ user: user_id })
-    const bdayPerson = await web.users.info({ user: BDAY_USER_ID })
-    const { user: { real_name: bdayPersonFullName } } = bdayPerson
-    const { user: { real_name: senderFullName } } = sender
 
-    const renderedText = (!matchesWithOptMessage)
-      ? 'Hope you have a wonderful day and eat lots of cake on behalf of all of us! :birthday:'
-      : matchesWithOptMessage['groups']['CUSTOM_MESSAGE']
-    return web.chat.postMessage({
-      channel: BDAY_USER_ID,
-      text: [
-        `:tada: Happy Birthday ${bdayPersonFullName}! :tada:`,
-        renderedText,
-        `Click :point_right: <${MIRO_URL}|here> :point_left: to see the birthday card!`,
-        `- From ${senderFullName} on behalf of EQ`
-      ].join('\n'),
-      blocks: [
-        {
-          'type': 'header',
-          'text': {
+    // when more fields are added
+    if (type === 'block_actions' && action.block_id === 'manage_fields') {
+      let updatedBlocks
+      let _buttons
+
```

Comment:
seem like a repetitive code block except for a few params, maybe could extract as a common function

Reply:
as it is a bit experimental, I'll leave it for now, but agree, could be transformed into a helper function.